Lib/test/test_pathlib.py

```diff
@@ -2478,6 +2478,226 @@ def test_complex_symlinks_relative(self):
     def test_complex_symlinks_relative_dot_dot(self):
         self._check_complex_symlinks(os.path.join('dirA', '..'))
 
+class WalkTests(unittest.TestCase):
+    """Tests for Path.walk()
+
+    They are mostly just copies of os.walk tests converted to use Paths
+    because os.walk tests are already mature and cover many edge cases
+    that we could miss writing new tests.
+
+    It is not combined with Path* tests because its setup is aimed at
+    testing a bit more complex cases than Path's setup.
+    """
+
+    # Wrapper to hide minor differences between different implementations
+    # of Path.walk (for example, Path.fwalk)
+    def walk(self, top, **kwargs):
+        return top.walk(**kwargs)
```

Comment:
IMO: needless indirection as we only have one implementation in pathlib.

Reply:
For now we do. What about the future? I mean, I'm okay with removing it tbh.